### PR TITLE
add email template and basic config setting

### PR DESCRIPTION
**Pull request by huojiao2006**

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -50,6 +51,11 @@ type MysqlConfig struct {
 	Disable  bool   `default:"false"`
 }
 
+type BasicConfig struct {
+	PlatformName string `default:"Openpitrix"`
+	PlatformUrl  string `default:"http://139.198.4.179:8000/"`
```
Comment (chilianyi) [inline pkg/config/config.go]: Openpitrix -> OpenPitrix
http://139.198.4.179:8000/ -> https://lab.openpitrix.io

File: pkg/constants/notify_message.go [hunk 2]
```diff
@@ -3,8 +3,7 @@ package constants
 import (
 	"bytes"
 	"fmt"
-	"html/template"
-	"strings"
+	"text/template"
```
Comment (chilianyi) [inline pkg/constants/notify_message.go]: Add copyright

File: pkg/constants/notify_message.go [hunk 3]
```diff
@@ -61,12 +47,21 @@ func (n *NotifyTitle) GetDefaultMessage(params ...interface{}) string {
 }
 
 func (n *NotifyContent) GetDefaultMessage(params ...interface{}) string {
-	t, _ := template.New(EmailNotifyName).Parse(EmailNotifyTemplate)
+	t, err := template.New(EmailNotifyName).Parse(EmailNotifyTemplate)
+	if err != nil {
+		fmt.Println("Parse email template string Failed:" + err.Error())
```
Comment (chilianyi) [inline pkg/constants/notify_message.go]: Change fmt to logger, Failed -> failed.

File: pkg/constants/notify_message.go [hunk 4]
```diff
@@ -61,12 +47,21 @@ func (n *NotifyTitle) GetDefaultMessage(params ...interface{}) string {
...
 	b := bytes.NewBuffer([]byte{})
 	emailContent := &EmailNotifyContent{
-		Content: strings.Split(n.GetMessage(defaultLocale, params...), "\n"),
+		Content: n.GetMessage(defaultLocale, params...),
 	}
-	t.Execute(b, emailContent)
+
+	err = t.Execute(b, emailContent)
+	if err != nil {
+		fmt.Println("Parse email template string Failed:" + err.Error())
```
Comment (chilianyi) [inline pkg/constants/notify_message.go]: Change fmt to logger, Failed -> failed.

File: pkg/service/account/handler_account.go [hunk 5]
```diff
@@ -586,15 +588,27 @@ func (p *Server) CreateUser(ctx context.Context, req *pb.CreateUserRequest) (*pb
 				Addresses:   []string{email},
 			})
 		} else {
+			platformName := pi.Global().GetBasicCfg().PlatformName
+			platformUrl := pi.Global().GetBasicCfg().PlatformUrl
+
+			accountClient, err := accountclient.NewClient()
```
Comment (chilianyi) [inline pkg/service/account/handler_account.go]: Here is account manager, should use imClient, accountclient is used for other manager.

File: pkg/service/account/handler_account.go [hunk 6]
```diff
@@ -664,9 +678,11 @@ func (p *Server) IsvCreateUser(ctx context.Context, req *pb.CreateUserRequest) (
 		if err != nil {
 			logger.Error(ctx, "Failed to get user [%s]: %+v", s.UserId, err)
 		} else {
+			platformName := pi.Global().GetBasicCfg().PlatformName
+			platformUrl := pi.Global().GetBasicCfg().PlatformUrl
 			emailNotifications = append(emailNotifications, &models.EmailNotification{
 				Title:       constants.IsvInviteMemberNotifyTitle.GetDefaultMessage(getUserResponse.User.Username),
-				Content:     constants.IsvInviteMemberNotifyContent.GetDefaultMessage(username, email, password),
+				Content:     constants.IsvInviteMemberNotifyContent.GetDefaultMessage(username, email, password, platformName, platformUrl),
```
Comment (chilianyi) [inline pkg/service/account/handler_account.go]: IsvInviteMemberNotifyContent have not add new parameters.

File: pkg/pi/pi.go [hunk 7]
```diff
@@ -161,3 +164,24 @@ func (p *Pi) openEtcd() *Pi {
...
+func (p *Pi) GetBasicCfg() *config.BasicConfig {
+	p.cfg = config.LoadConf()
+	basicConfig := &config.BasicConfig{
+		PlatformName: p.cfg.Basic.PlatformName,
+		PlatformUrl:  p.cfg.Basic.PlatformUrl,
+	}
+	return basicConfig
+}
+
+func (p *Pi) SetBasicCfg(basicConfig config.BasicConfig) error {
```
Comment (chilianyi) [inline pkg/pi/pi.go]: We often use pointer here
func (p *Pi) SetBasicCfg(basicConfig *config.BasicConfig) error {

File: pkg/pi/pi.go [hunk 8]
```diff
@@ -161,3 +164,24 @@ func (p *Pi) openEtcd() *Pi {
...
+	if !govalidator.IsURL(basicConfig.PlatformUrl) {
+		return gerr.New(nil, gerr.InvalidArgument, gerr.ErrorIllegalUrlFormat, basicConfig.PlatformUrl)
+	}
+	mutex.Lock()
+	os.Setenv("OPENPITRIX_BASIC_PLATFORM_NAME", basicConfig.PlatformName)
+	os.Setenv("OPENPITRIX_BASIC_PLATFORM_URL", basicConfig.PlatformUrl)
+	config.LoadConf()
```
Comment (chilianyi) [inline pkg/pi/pi.go]: Put global config into etcd, not just system environment. Should not lose when restart container.

File: pkg/pi/pi.go [hunk 9]
```diff
@@ -161,3 +164,24 @@ func (p *Pi) openEtcd() *Pi {
 	p.etcd = e
 	return p
 }
+
+func (p *Pi) GetBasicCfg() *config.BasicConfig {
+	p.cfg = config.LoadConf()
```
Comment (chilianyi) [inline pkg/pi/pi.go]: Get basic config from etcd.

File: api/service_config.proto [hunk 10]
```diff
@@ -36,9 +37,15 @@ message RuntimeConfig {
...
 
+message BasicConfig {
+	google.protobuf.StringValue platform_name = 1;
+	google.protobuf.StringValue platform_url = 2;
+}
+
 message SetServiceConfigRequest {
 	NotificationConfig notification_config = 1;
 	RuntimeConfig runtime_config = 2;
+	BasicConfig basic_config=3;
```
Comment (chilianyi) [inline api/service_config.proto]: Need fmt.

File: api/service_config.proto [hunk 11]
```diff
@@ -52,6 +59,7 @@ message GetServiceConfigRequest {
 message GetServiceConfigResponse {
 	NotificationConfig notification_config = 1;
 	RuntimeConfig runtime_config = 2;
+	BasicConfig basic_config=3;
```
Comment (chilianyi) [inline api/service_config.proto]: Need fmt.

File: api/service_config.proto [hunk 12]
```diff
@@ -23,6 +23,7 @@ message EmailServiceConfig {
 	google.protobuf.BoolValue ssl_enable = 7;
 }
 
+
```
Comment (chilianyi) [inline api/service_config.proto]: Useless empty line.

File: pkg/constants/notify_message.go (hunk 2 above)
Comment (huojiao2006) [reply pkg/constants/notify_message.go]: done

**Pull request closed without merging**